pkg/portal/appresource/manager.go

```diff
@@ -299,6 +300,11 @@ func (m *Manager) applyUpdates(appID string, appFs resource.Fs, updates []Update
 			return nil, nil, err
 		}
 
+		if u.Checksum != "" && crypto.ChecksumString(resrc.Data) != u.Checksum {
+			err = fmt.Errorf("checksum not equal '%s'", u.Path)
```

Comment:
Define a new API error in `pkg/portal/appresource/error.go`. It is because we use API error to represent error that we anticipate to occur.

```
var ResourceUpdateConflict = apierrors.DataRace.WithReason("ResourceUpdateConflict")
```

And use it like

```
msg := fmt.Sprintf("resource update conflict: %v", path)
err = ResourceUpdateConflict.NewWithInfo(msg,  apierrors.Details{"path": path})
```